### ISSUE TITLE
[BUG] KOKKOS version of fix dt/reset sets timestep incorrectly in most cases

### ISSUE THREAD

**Issue by mkanski** [lammps/lammps#3961]
**Summary**

The KOKKOS version of `fix dt/reset` sets a larger timestep than the standard implementation in most cases.

**LAMMPS Version and Platform**

Current develop, Linux Mint 21.3

**Expected Behavior**

The timestep should be the same in both versions.

**Actual Behavior**

The timestep is higher than it should be in almost all cases. It only works if the fastest atom has the very last id. The input file below shows the issue. If you change `id 5` to `id 10`, the code correctly sets the timestep.

**Steps to Reproduce**

Compile LAMMPS with KOKKOS and run the following input in serial mode:

```
units real
region test sphere 0 0 0 30
create_box 1 test
mass 1 1
create_atoms 1 random 10 235 test
group move id 5
velocity move set 100 0 0 units box
fix dt all dt/reset 1 0.0 0.1 0.0000001 units box
thermo_style custom step dt
run 0
```

**Further Information, Files, and Links**

This is a little bit embarrassing, as I've coded the KOKKOS implementation of `fix dt/reset`. After some debugging I found that the problem lies in the `parallel_reduce` function, which calculates the timestep for each atom, but only uses its value for the last atom when looking for the smallest timestep.
This issue went unnoticed through my testing, because the fastest atom had always the largest id.

Had I misunderstood how `parallel_reduce` works, or is there bug somewhere there? I have a working bugfix, but it's less elegant than just using `parallel_reduce`, so I'd like to ask first if the current code can be corrected.

**Comment by stanmoore1** [comment]
@mkanski thanks for the report, will take a look.

**Comment by stanmoore1** [comment]
@mkanski this should be fixed with 8c3cda0, just needed to return the minimum timestep inside the functor. Also class variables (e.g. `emax`) in stack memory are accessible on the device, so no need to copy them into a view.

**Comment by mkanski** [comment]
Thanks for the fast help! :)
